### ISSUE TITLE
[BUG] Harvester Cloud Provider is not able to deploy upgraded container after upgrading the cluster

### ISSUE THREAD

**Issue by slickwarren** [harvester/harvester#2546]
**Rancher Server Setup**
- Rancher version: v2.6.7-rc3
- Installation option (Docker install/Helm Chart): docker
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: v1.23.8+rke2r2 -> 1.23.9-rc3+rke2r1
- Cluster Type (Local/Downstream): downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): harvester

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom): standard user, cluster owner

**Describe the bug**
upgrade from 1.23.x -> rc’d version updates the chart HOWEVER the deployment is not successfully deploying the new container with version 0.1.4

**To Reproduce**
* deploy a harvester downstream cluster through rancher using the node driver and k8s 1.23.8
  * observe the old versions of the chart are used
* change the KDM to one that has the RC'd versions available
* upgrade the cluster to 1.23.x-rc3

**Result**
the harvester cloud provider container is not updating properly

**Expected Result**

Chart workloads and pods should successfully upgrade

**Screenshots**
![Screen Shot 2022-07-19 at 12 10 40 PM](https://user-images.githubusercontent.com/16691014/179852381-8f933371-727a-47a7-847c-66546630f278.png)

**Additional context**

fresh install of rc version is working fine.

**Comment by irishgordo** [comment]
@slickwarren I was able to also see that the `harvester-cloud-provider-UUID` container was stuck in a `ContainerCreating` state.

Example:
```
Name:         harvester-cloud-provider-5d49884d84-vp7jz                                                                                                                                                         
Namespace:    kube-system                                                                                                                                                                                       
Priority:     0                                                                                                                                                                                                 
Node:         caleb-test-cluster-pool1-9b77ad6a-q4ltt/192.168.10.50                                                                                                                                             
Start Time:   Wed, 20 Jul 2022 10:55:45 -0700                                                                                                                                                                   
Labels:       app.kubernetes.io/instance=harvester-cloud-provider                                                                                                                                               
              app.kubernetes.io/name=harvester-cloud-provider                                                                                                                                                   
              pod-template-hash=5d49884d84                                                                                                                                                                      
Annotations:  kubernetes.io/psp: global-unrestricted-psp                                                                                                                                                        
Status:       Pending                                                                                                                                                                                           
IP:           192.168.10.50                                                                                                                                                                                     
IPs:                                                                                                                                                                                                            
  IP:           192.168.10.50                                                                                                                                                                                   
Controlled By:  ReplicaSet/harvester-cloud-provider-5d49884d84                                                                                                                                                  
Containers:                                                                                                                                                                                                     
  harvester-cloud-provider:                                                                                                                                                                                     
    Container ID:                                                                                                                                                                                               
    Image:         rancher/harvester-cloud-provider:v0.1.4                                                                                                                                                      
    Image ID:                                                                                                                                                                                                   
    Port:          <none>                                                                                                                                                                                       
    Host Port:     <none>                                                                                                                                                                                       
    Command:                                                                                                                                                                                                    
      harvester-cloud-provider                                                                                                                                                                                  
    Args:                                                                                                                                                                                                       
      --cloud-config=/etc/kubernetes/cloud-config                                                                                                                                                               
    State:          Waiting                                                                                                                                                                                     
      Reason:       ContainerCreating                                                                                                                                                                           
    Ready:          False                                                                                                                                                                                       
    Restart Count:  0                                                                                                                                                                                           
    Environment:    <none>                                                                                                                                                                                      
    Mounts:                                                                                                                                                                                                     
      /etc/kubernetes/cloud-config from cloud-config (rw)                                                                                                                                                       
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-x6p8b (ro)                                                                                                                             
Conditions:                                                                                                                                                                                                     
  Type              Status                                                                                                                                                                                      
  Initialized       True                                                                                                                                                                                        
  Ready             False                                                                                                                                                                                       
  ContainersReady   False                                                                                                                                                                                       
  PodScheduled      True                                                                                                                                                                                        
Volumes:                                                                                                                                                                                                        
  cloud-config:                                                                                                                                                                                                 
    Type:          HostPath (bare host directory volume)                                                                                                                                                        
    Path:          /etc/kubernetes/cloud-config                                                                                                                                                                 
    HostPathType:  File                                                                                                                                                                                         
  kube-api-access-x6p8b:                                                                                                                                                                                        
    Type:                    Projected (a volume that contains injected data from multiple sources)                                                                                                             
    TokenExpirationSeconds:  3607                                                                                                                                                                               
    ConfigMapName:           kube-root-ca.crt                                                                                                                                                                   
    ConfigMapOptional:       <nil>                                                                                                                                                                              
    DownwardAPI:             true                                                                                                                                                                               
QoS Class:                   BestEffort                                                                                                                                                                         
Node-Selectors:              kubernetes.io/os=linux                                                                                                                                                             
Tolerations:                 cattle.io/os=linux:NoSchedule                                                                                                                                                      
                             node-role.kubernetes.io/control-plane:NoSchedule                                                                                                                                   
                             node-role.kubernetes.io/etcd:NoExecute                                                                                                                                             
                             node.cloudprovider.kubernetes.io/uninitialized=true:NoSchedule                                                                                                                     
                             node.kubernetes.io/not-ready:NoExecute op=Exists for 300s                                                                                                                          
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s                                                                                                                        
Events:                                                                                                                                                                                                         
  Type     Reason       Age                   From     Message                                                                                                                                                  
  ----     ------       ----                  ----     -------                                                                                                                                                  
  Warning  FailedMount  9m16s (x21 over 63m)  kubelet  Unable to attach or mount volumes: unmounted volumes=[cloud-config], unattached volumes=[cloud-config kube-api-access-x6p8b]: timed out waiting for the c
ondition                                                                                                                                                                                                        
  Warning  FailedMount  4m48s (x5 over 50m)   kubelet  Unable to attach or mount volumes: unmounted volumes=[cloud-config], unattached volumes=[kube-api-access-x6p8b cloud-config]: timed out waiting for the c
ondition                                                                                                                                                                                                        
  Warning  FailedMount  39s (x40 over 65m)    kubelet  MountVolume.SetUp failed for volume "cloud-config" : hostPath type check failed: /etc/kubernetes/cloud-config is not a file                              
                                                                                                                                                                                      
```

## Setup:
### Rancher Node:
SuperMicro Bare Metal Intel Atom, 4 Core, 4 GiB DDR3 RAM
Rancher Version, installed with Helm: **v2.6.7-rc3**
### Harvester Node:
Dell PowerEdge R720, Intel Xeon 20 Core, 64 GiB DDR3 ECC RAM
Harvester Version: **master-aa3d95d2-head**

Was able to reproduce following steps.
See attached photos.
![Screenshot from 2022-07-20 11-46-06](https://user-images.githubusercontent.com/5370752/180061898-d4e6f8e0-4dc1-4a3b-8668-18794bf516cb.png)
![Screenshot from 2022-07-20 11-43-13](https://user-images.githubusercontent.com/5370752/180061906-2d83f9ef-029c-4612-8422-cf73ea7c070d.png)
![Screenshot from 2022-07-20 11-40-05](https://user-images.githubusercontent.com/5370752/180061910-409770f3-c46e-4ad1-b6d9-c75e70a176be.png)
![Screenshot from 2022-07-20 11-35-33](https://user-images.githubusercontent.com/5370752/180061912-b3b18fdb-d19c-404e-81a5-eceeaef8cb39.png)
![trying-for-logs-on-container-creating](https://user-images.githubusercontent.com/5370752/180061916-b6dc2ca9-5937-4878-83f9-c44e860c5ddb.png)
![Screenshot from 2022-07-20 11-14-00](https://user-images.githubusercontent.com/5370752/180061919-32c38868-be53-4273-a0b8-fae0505b1fbd.png)
![rolling-out-plans-for-v123-9-rc3](https://user-images.githubusercontent.com/5370752/180061923-89b05095-24d4-450b-bbd7-2943bcc3630d.png)
![upgrading_to_123 9-rc3](https://user-images.githubusercontent.com/5370752/180061924-637946d0-2c6b-4cc2-a5ea-1051c7e57152.png)
![refreshing_metadata](https://user-images.githubusercontent.com/5370752/180061926-fc2fc0b5-d52a-4e2f-b6d1-8f00358f1b29.png)
![modified_rke_metadata](https://user-images.githubusercontent.com/5370752/180061930-719a5ea0-e077-4ebe-8a5b-4e31245a8434.png)
![original_rke_metadata](https://user-images.githubusercontent.com/5370752/180061933-b91f3ca4-c590-416e-8976-fee893d84dd4.png)
![v1_23_8_no_rc_setup_3_node](https://user-images.githubusercontent.com/5370752/180061935-b29724cf-c1f5-4aa1-989b-2187847513da.png)

It does look like at least in one view that the cluster is on the RC version, but that contianer creating is just hung.

Also attached is the support bundle from Harvester
[supportbundle_ca587245-2aaa-4820-a6ab-fbb04824a0a4_2022-07-20T18-41-42Z.zip](https://github.com/harvester/harvester/files/9153785/supportbundle_ca587245-2aaa-4820-a6ab-fbb04824a0a4_2022-07-20T18-41-42Z.zip)

**Comment by chrisho** [comment]
After testing, we found that the Cluster Config UI page will submit `cloudPathConfig` to the backend when adding a guest cluster, and this request parameter is missing when editing, the frontend is being fixed.

add:
![image](https://user-images.githubusercontent.com/3347460/180376442-39414e4a-39cf-466f-9b63-fbd4d574d3a3.png)

edit:
![image](https://user-images.githubusercontent.com/3347460/180376461-5158d76c-82b8-414c-8ab6-dc60666cdfec.png)

**Comment by TachunLin** [comment]
Verified **fixed** on Harvester `v1.0.2 GA` with Rancher `v2.6.7-rc3`. Close this issue.

### Result
After upgrading the existing RKE2 guest cluster kubernetes version from older `v1.23.8+rke2r1` to `1.23.9-rc3+rke2r1`.
The Harvester cloud provider can successfully updated from `v0.1.3` to `v0.1.4`  

Cloud provider before upgrade: v0.1.3
  ![image](https://user-images.githubusercontent.com/29251855/180738877-56afcd55-e519-48d9-a8b8-3cbed91a1dfb.png)

After upgrade: v0.1.4
  ![image](https://user-images.githubusercontent.com/29251855/180740998-80e451e5-ad91-4111-8abe-f51395427b9c.png)

### Test Information
* Test Environment: 1 node harvester on dolphin bare machine
* Harvester version: `v1.0.2-GA`
* Rancher version: `v2.6.7-rc3`
* ui-dashboard-index: https://releases.rancher.com/dashboard/latest/index.html
* ui-offline-preferred: Remote
* rke-metadata-config: 
  ```
  {"refresh-interval-minutes":"1440","url":"https://yufa-dev.s3.ap-east-1.amazonaws.com/data.json"}
  ```

### Verify Steps
1. Prepare Rancher v2.6.7-rc3 and Harvester v1.0.2 GA
2. Update `rke-metadata-config` to `{"refresh-interval-minutes":"1440","url":"https://yufa-dev.s3.ap-east-1.amazonaws.com/data.json"}` in global settings
  ![image](https://user-images.githubusercontent.com/29251855/180735267-939e92e3-7fd5-4659-8bc8-ab14c95161d8.png)
3. Update the ui-dashboard-index to `https://releases.rancher.com/dashboard/latest/index.html`
4. Set `ui-offline-preferred` to `Remote`
5. Refresh web page (ctrl + r)
6. Open Create RKE2 cluster page
7.  Check the `show deprecated kubernetes patched versions`
  ![image](https://user-images.githubusercontent.com/29251855/180736528-feaa9615-ccf9-482b-9354-c2c9a6a4b23b.png)
8. Select `v1.23.8+rke2r1` 
9. Finish the RKE2 cluster provision 
  ![image](https://user-images.githubusercontent.com/29251855/180738516-3f429bba-22ab-4476-bebf-0ac2f87935c3.png)
10. Check the current cloud provider version in workload page
  ![image](https://user-images.githubusercontent.com/29251855/180738877-56afcd55-e519-48d9-a8b8-3cbed91a1dfb.png)
11. Edit RKE2 cluster, upgrade the kubernetes version to `1.23.9-rc3+rke2r1`
  ![image](https://user-images.githubusercontent.com/29251855/180739231-e61ef680-5a9d-480b-9ac9-eda7839e17b6.png)
  ![image](https://user-images.githubusercontent.com/29251855/180739331-611b05d4-0c5d-4835-9da0-8c05b9cca027.png)
12. Wait for update finish
  ![image](https://user-images.githubusercontent.com/29251855/180739876-dc409fa8-a9a6-406b-a614-085cea57121f.png)
13. The cloud provider is upgrading
  ![image](https://user-images.githubusercontent.com/29251855/180740637-5d1c6ce0-07ed-4a62-a364-f1b5e9fe473f.png)
14. delete the old cloud provider version pod (v0.1.3)
  ![image](https://user-images.githubusercontent.com/29251855/180740767-e6d5cdc2-c004-4c7a-8298-690775265002.png)
15. Wait for newer version cloud provider have been bumped
  ![image](https://user-images.githubusercontent.com/29251855/180740875-38fa0cc0-c13a-4e39-ba46-5e869eadf087.png)
  ![image](https://user-images.githubusercontent.com/29251855/180740998-80e451e5-ad91-4111-8abe-f51395427b9c.png)
